packages/@aws-cdk/core/lib/arn.ts

```diff
@@ -130,10 +130,15 @@ export class Arn {
    * the 'scope' is attached to. If all ARN pieces are supplied, the supplied scope
    * can be 'undefined'.
    */
-  public static format(components: ArnComponents, stack: Stack): string {
-    const partition = components.partition ?? stack.partition;
-    const region = components.region ?? stack.region;
-    const account = components.account ?? stack.account;
+  public static format(components: ArnComponents, stack?: Stack): string {
+    const partition = components.partition ?? stack?.partition;
+    const region = components.region ?? stack?.region;
+    const account = components.account ?? stack?.account;
+
+    if (partition === undefined || region === undefined || account === undefined) {
```

Comment:
Test for this?

Also, what about `null` or empty strings?

Reply:
Ack.

Empty string is fine, I'll catch `null` as well.